core/src/core.cu

```diff
@@ -437,6 +437,14 @@ inline void registerParameters()
     AMG_Config::registerParameter<int>("kaczmarz_coloring_needed", "Enforces MC Kaczmarz (0 for naive single warp implementation)", 1);
     //CF-Jacobi smoother
     AMG_Config::registerParameter<int>("cf_smoothing_mode", "Flavour of CF-smoothing. 0=CF for the presmoothing and FC for post smoothing. 1=opposite of 0. Not yet implemented: 2=FCF. 3=CFC", 0);
+    //Multicolor Gauss-Seidel
+    std::vector<std::string> gs_values;
+    gs_values.push_back("DEFAULT");
+    gs_values.push_back("4_PER_ROW");
+    gs_values.push_back("32_PER_ROW");
+    gs_values.push_back("WARP_PER_ROW");
+    gs_values.push_back("NAIVE");
+    AMG_Config::registerParameter<std::string>("gs_method", "Gauss-Seidel kernel <DEFAULT,4_PER_ROW,32_PER_ROW,WARP_PER_ROW,NAIVE>", "DEFAULT", gs_values);

```

Comment:
Is there some guidance on which parameter(s) is best given some particular characteristic of the problem?

Reply:
Ok, that one i am little bit doubtful about. Parameter was added for development purposes to be able to switch different kernels and compare them. Default value is "default" and it will enable simple heuristic that will try to choose best parameter for given matrix in smoother - you can look at this heuristic as a guidance on this parameter. However there is incentive to hide this parameter from the user and always use heuristic - less confusion overall about parameters usage. What do you think?

Reply:
Thanks - I didn't spot the heuristic in the default setting yet, that's great. 

I think it is OK to expose the parameter if we expect there is some case where the heuristic is insufficient, otherwise I think hiding is preferable to avoid needlessly increasing the user parameter search space.